modules/release-notes/con-breaking-changes.adoc

```diff
@@ -24,3 +26,10 @@ auth:
     guest:
       dangerouslyAllowOutsideDevelopment: true
 ```
+
+== Improved validation permission policies from different sources
+In this release, {product-short} provides more strict validation on the source of permission policies and roles based on how you define the first role.
+
+This update improves the validation of the different sources of permission policies and roles and provides more consistent policy definition. If a permission policy or role with a new member does not match the originating role's source, {product-short} prevents any update to permissions. Sources include 'REST, 'CSV', 'Configuration', and 'legacy'.
+
+Before updating your {product} application, you should migrate all permission policies and roles to a single source by using the GET roles endpoint to review the source information and by querying the role-metadata table of the permission database. You can make updates to permission policies by using one of the following: REST API, CSV file, and the database.
```

Comment:
I would say OR would be fine. Any of those options are potentially valid methods for updating the roles and permissions. 

And to give some context to what was changed. We improved upon the validation of the permissions and roles to try to have more consistency throughout the RBAC plugin. This validation now ensures that permission policies and matching roles all originate from the same source ('REST', 'CSV', 'Configuration', and 'Legacy').

An example would be the role `role:default/team-a` being defined in the CSV file. All permissions policies associated with that role and any members that are added to that role should be done from the CSV file.

`user:default/tony`, `user:default/peter`, and `user:default/bruce` was added to `role:default/team-a` through the CSV file. And permissions for this role also were created in the CSV file.

The original problem that we were trying to solve is that some permission policies and roles could be mismatched. Say for example user `user:default/tom` is attached to `role:default/team-b` that was created using the REST API but then there is also user `user:default/adam` that is also attached to `role:default/team-b` within the CSV file that was created first. 

This inconsistency leads to problems because users might try to delete the `user:default/adam` from 
`role:default/team-b` using the REST API only for it to come back since this example was created using the CSV file.

For a method to fix this problem, users can query the role and permission information using the `GET` endpoints of the two. Then they can make updates using either the CSV file, REST API, Config file, or the database by removing the conflicting permissions and roles and then adding them back using the appropriate source.